src/analysis/ml.py

```diff
@@ -289,3 +287,21 @@ def hac_horizontal(dataframe: pd.DataFrame, output_png: str, output_file: str, l
     clusters_df.to_csv(output_file, sep='\t', index=False)
     make_required_dirs(output_png)
     plt.savefig(output_png, bbox_inches="tight", dpi=DPI)
+
+def ensemble_network(dataframe: pd.DataFrame, output_file:str):
+    """
+    this method will calculate the mean of the binary values in the provided dataframe
+    -  we're counting the number of times an edge appears in a set of pathway reconstruction networks / by the length of the set
+    edges that appear more frequently will have a higher number, thus are more likey to be robust,
+    so this information can be used to include an edge to include in a final network
+    @param datafram: binary dataframe of edge comparison between algorithms from summarize_networks
+    @param output_file: the file name to save the ensemble network
+    """
+    row_means = np.mean(dataframe, axis=1).reset_index()
+    row_means.columns = ['Edges', 'Frequency']
+    row_means[['Node1', 'Node2']] = row_means['Edges'].str.split('\|\|\|',expand=True)
```

Comment:
```suggestion
    row_means[['Node1', 'Node2']] = row_means['Edges'].str.split(NODE_SEP, expand=True)
```
Will this still work? I'd like to use the constant `NODE_SEP` in case we ever change it.

Reply:
The problem with using '|||' is that in the str.split is that it is treating the each character in the string as a separate delimiter. Also the '|' means or, so we needed to escape it.

Reply:
I tested this with `regex=False`. Initially I had an old version of pandas that didn't support that argument, but when I updated to pandas 1.4.4. (the version in the SPRAS environment) it worked
```
> df = pd.DataFrame({'Edges': ['A|||B', 'C|||D', 'E|||F']})
> df['Edges'].str.split('|||', regex=False, expand=True)
   0  1
0  A  B
1  C  D
2  E  F
```

I still didn't test it here, but it looks closer to what we want.

Reply:
It worked once I updated my pandas as well